frontend/class-twitter.php

```diff
@@ -375,15 +374,19 @@ private function gallery_images_output() {
 	 * @return bool
 	 */
 	private function taxonomy_image_output() {
+		$img = false;
 		foreach ( array( 'twitter-image', 'opengraph-image' ) as $tag ) {
 			$img = WPSEO_Taxonomy_Meta::get_meta_without_term( $tag );
-			if ( $img !== '' ) {
-				$this->image_output( $img );
-
-				return true;
+			if ( $img && $img !== '' ) {
+				break;
```

Comment:
You are changing the behaviour of this method. The issue is about setting an image if there isn't one set yet. In this situation this method might possibly change a set image. I am not sure if this is intended. /cc @moorscode @atimmer

Reply:
Hi @R33D3M33R 

The additional functionality you are implementing here is going against the functionality we want to provide.
When `twitter-image` or `opengraph-image` are set, these are configured manually and intentionally by the user.

If a filter allows to override these images with another image, the expectations are not met and unwanted situations will occur.

We really understand the situation and need you describe in your original issue, in which it is impossible to set the image when no image has been entered by the user in the WordPress admin area.

The suggested code that @andizer shared with you was keeping the current (and intended) functionality in tact and only adds the extra filter whenever no image was set directly. This limits the situations you can programmatically alter the image that will be rendered in the content, though this limitation is a desired one.